tests/model_fields/models.py

```diff
@@ -687,3 +687,27 @@ class Meta:
                 F("a"), name="Generated model unique constraint virtual a"
             ),
         ]
+
+
+class GeneratedModelChoices(models.Model):
+    c = models.IntegerField(choices=Whiz.CHOICES, null=True)
+    c_squared = models.GeneratedField(
+        expression=F("c") * F("c"),
+        output_field=models.IntegerField(choices=Whiz.CHOICES, null=True),
+        db_persist=True,
+    )
```

Comment:
If we support multiple sources of choices, we should test the cases of GeneratedField(choices=FOO, output_field=Field(choices=BAR)) where the choices are different so there's a test confirming which is taking precedence.